cmd/earthly/main.go

```diff
@@ -2821,7 +2821,7 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
 	// Default upload logs, unless explicitly configured
 	if !app.cfg.Global.DisableLogSharing {
 		_, _, _, whoAmIErr := cc.WhoAmI()
-		isLoggedIn := err == nil
+		isLoggedIn := whoAmIErr == nil
```

Comment:
@brandonSc This is why you managed to get an unauthorized. Fixed now.

Reply:
So we shouldn't even get the error when not logged in. I think that invalidates my other comment below.

Reply:
Yeah. I was super confused and then realized my mistake :facepalm: